Ansible bump to 2.15.3 causing issue with ensure-ansible.sh

**What steps did you take and what happened:**

if ansible is not installed and python is v3.8 (like in ubuntu 2004 LTS)
PR #1275 is causing issue when ensure-ansible.sh try to install ansible

ensure-ansible.sh failed with

```
hack/ensure-ansible.sh
fatal: No names found, cannot describe anything.
ERROR: Could not find a version that satisfies the requirement ansible-core==2.15.3 (from versions: 0.0.1a1, 2.11.0b1, 2.11.0b2, 2.11.0b3, 2.11.0b4, 2.11.0rc1, 2.11.0rc2, 2.11.0, 2.11.1rc1, 2.11.1, 2.11.2rc1, 2.11.2, 2.11.3rc1, 2.11.3, 2.11.4rc1, 2.11.4, 2.11.5rc1, 2.11.5, 2.11.6rc1, 2.11.6, 2.11.7rc1, 2.11.7, 2.11.8rc1, 2.11.8, 2.11.9rc1, 2.11.9, 2.11.10rc1, 2.11.10, 2.11.11rc1, 2.11.11, 2.11.12rc1, 2.11.12, 2.12.0b1, 2.12.0b2, 2.12.0rc1, 2.12.0, 2.12.1rc1, 2.12.1, 2.12.2rc1, 2.12.2, 2.12.3rc1, 2.12.3, 2.12.4rc1, 2.12.4, 2.12.5rc1, 2.12.5, 2.12.6rc1, 2.12.6, 2.12.7rc1, 2.12.7, 2.12.8rc1, 2.12.8, 2.12.9rc1, 2.12.9, 2.12.10rc1, 2.12.10, 2.13.0b0, 2.13.0b1, 2.13.0rc1, 2.13.0, 2.13.1rc1, 2.13.1, 2.13.2rc1, 2.13.2, 2.13.3rc1, 2.13.3, 2.13.4rc1, 2.13.4, 2.13.5rc1, 2.13.5, 2.13.6rc1, 2.13.6, 2.13.7rc1, 2.13.7, 2.13.8rc1, 2.13.8, 2.13.9rc1, 2.13.9, 2.13.10rc1, 2.13.10, 2.13.11rc1, 2.13.11)
ERROR: No matching distribution found for ansible-core==2.15.3
make: *** [Makefile:174: deps-all] Error 1
```

trying to reproduce manually we see that there is another error who explain the problem

```
ERROR: Ignored the following versions that require a different python version:  2.15.3 Requires-Python >=3.9; 
```


/kind bug
[One or more /area label. See https://github.com/kubernetes-sigs/cluster-api/labels?q=area for the list of labels]

--- a/images/capi/Makefile
+++ b/images/capi/Makefile
@@ -53,67 +53,60 @@ version: ## Display version of image-builder
 deps: ## Installs/checks all dependencies
 deps: deps-ami deps-azure deps-do deps-gce deps-ova deps-openstack deps-qemu deps-raw deps-oci deps-osc deps-vbox deps-powervs deps-nutanix deps-hcloud
 
+.PHONY: deps-common
+deps-common: ## Installs/checks dependencies common to most builds
+deps-common:
+	hack/ensure-python.sh
+	hack/ensure-ansible.sh
+	hack/ensure-packer.sh
+
 .PHONY: deps-ami
 deps-ami: ## Installs/checks dependencies for AMI builds
-deps-ami:
-	hack/ensure-ansible.sh
+deps-ami: deps-common
 	hack/ensure-ansible-windows.sh
-	hack/ensure-packer.sh
 	hack/ensure-goss.sh
 	$(PACKER) init packer/config.pkr.hcl
 
 .PHONY: deps-azure
 deps-azure: ## Installs/checks dependencies for Azure builds
-deps-azure:
-	hack/ensure-ansible.sh
+deps-azure: deps-common
 	hack/ensure-ansible-windows.sh
-	hack/ensure-packer.sh
+	hack/ensure-goss.sh
 	hack/ensure-jq.sh
 	hack/ensure-azure-cli.sh
-	hack/ensure-goss.sh
 	$(PACKER) init packer/config.pkr.hcl
 	$(PACKER) init packer/azure/config.pkr.hcl
 
 .PHONY: deps-do
 deps-do: ## Installs/checks dependencies for DigitalOcean builds
-deps-do:
-	hack/ensure-ansible.sh
-	hack/ensure-packer.sh
+deps-do: deps-common
 	$(PACKER) init packer/config.pkr.hcl
 	$(PACKER) init packer/digitalocean/config.pkr.hcl
 
 .PHONY: deps-osc
 deps-osc: ## Installs/checks dependencies for Outscale builds
-deps-osc:
-	hack/ensure-ansible.sh
-	hack/ensure-packer.sh
+deps-osc: deps-common
 	hack/ensure-goss.sh
 	$(PACKER) init packer/config.pkr.hcl
 	$(PACKER) plugins install github.com/outscale/outscale
 
 .PHONY: deps-gce
 deps-gce: ## Installs/checks dependencies for GCE builds
-deps-gce:
-	hack/ensure-ansible.sh
-	hack/ensure-packer.sh
+deps-gce: deps-common
 	hack/ensure-goss.sh
 	$(PACKER) init packer/config.pkr.hcl
 
 .PHONY: deps-ova
 deps-ova: ## Installs/checks dependencies for OVA builds
-deps-ova:
-	hack/ensure-ansible.sh
+deps-ova: deps-common
 	hack/ensure-ansible-windows.sh
-	hack/ensure-packer.sh
 	hack/ensure-goss.sh
 	hack/ensure-ovftool.sh
 	$(PACKER) init packer/config.pkr.hcl
 
 .PHONY: deps-openstack
 deps-openstack: ## Installs/checks dependencies for OpenStack builds
-deps-openstack:
-	hack/ensure-ansible.sh
-	hack/ensure-packer.sh
+deps-openstack: deps-common
 	hack/ensure-goss.sh
 	hack/ensure-s3.sh
 	$(PACKER) init packer/config.pkr.hcl
@@ -121,43 +114,33 @@ deps-openstack:
 
 .PHONY: deps-qemu
 deps-qemu: ## Installs/checks dependencies for QEMU builds
-deps-qemu:
-	hack/ensure-ansible.sh
-	hack/ensure-packer.sh
+deps-qemu: deps-common
 	hack/ensure-goss.sh
 	$(PACKER) init packer/config.pkr.hcl
 
 .PHONY: deps-raw
 deps-raw: ## Installs/checks dependencies for RAW builds
-deps-raw:
-	hack/ensure-ansible.sh
-	hack/ensure-packer.sh
+deps-raw: deps-common
 	hack/ensure-goss.sh
 	$(PACKER) init packer/config.pkr.hcl
 
 .PHONY: deps-oci
 deps-oci: ## Installs/checks dependencies for OCI builds
-deps-oci:
-	hack/ensure-ansible.sh
-	hack/ensure-packer.sh
+deps-oci: deps-common
 	hack/ensure-ansible-windows.sh
 	$(PACKER) init packer/config.pkr.hcl
 	$(PACKER) plugins install github.com/hashicorp/oracle
 
 .PHONY: deps-vbox
 deps-vbox: ## Installs/checks dependencies for VirtualBox builds
-deps-vbox:
-	hack/ensure-ansible.sh
+deps-vbox: deps-common
 	hack/ensure-ansible-windows.sh
-	hack/ensure-packer.sh
 	hack/ensure-goss.sh
 	$(PACKER) init packer/config.pkr.hcl
 
 .PHONY: deps-powervs
 deps-powervs: ## Installs/checks dependencies for PowerVS builds
-deps-powervs:
-	hack/ensure-ansible.sh
-	hack/ensure-packer.sh
+deps-powervs: deps-common
 	hack/ensure-goss.sh
 	hack/ensure-powervs.sh
 	$(PACKER) init packer/config.pkr.hcl
@@ -170,9 +153,7 @@ deps-ignition:
 
 .PHONY: deps-nutanix
 deps-nutanix: ## Installs/checks dependencies for Nutanix builds
-deps-nutanix:
-	hack/ensure-ansible.sh
-	hack/ensure-packer.sh
+deps-nutanix: deps-common
 	hack/ensure-goss.sh
 	$(PACKER) init packer/config.pkr.hcl
 	$(PACKER) init packer/nutanix/config.pkr.hcl
@@ -186,9 +167,7 @@ deps-release: ## Installs/checks dependencies for project releases
 
 .PHONY: deps-hcloud
 deps-hcloud: ## Installs/checks dependencies for Hetznercloud builds
-deps-hcloud:
-	hack/ensure-ansible.sh
-	hack/ensure-packer.sh
+deps-hcloud: deps-common
 	hack/ensure-goss.sh
 	$(PACKER) init packer/config.pkr.hcl
 	$(PACKER) init packer/hcloud/config.pkr.hcl

--- a/images/capi/hack/ensure-python.sh
+++ b/images/capi/hack/ensure-python.sh
@@ -1,0 +1,51 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+[[ -n ${DEBUG:-} ]] && set -o xtrace
+
+# Python 3.9 or later is required, specifically for Ansible 2.14 or later.
+minimum_python_version=3.9.0
+
+# Ensure that Python exists and is a viable version.
+verify_python_version() {
+  if [[ -z "$(command -v python3)" ]]; then
+    cat <<EOF
+Can't find 'python3' in PATH, please fix and retry.
+See https://www.python.org/downloads/ for installation instructions.
+EOF
+    return 2
+  fi
+
+  local python_version
+  IFS=" " read -ra python_version <<< "$(python3 --version)"
+  if [[ "${minimum_python_version}" != $(echo -e "${minimum_python_version}\n${python_version[1]}" | sort -s -t. -k 1,1 -k 2,2n -k 3,3n | head -n1) && "${python_version[1]}" != "devel" ]]; then
+    cat <<EOF
+Detected python version: ${python_version[*]}.
+Ansible requires ${minimum_python_version} or greater.
+Please install ${minimum_python_version} or later.
+EOF
+    return 2
+  fi
+}
+
+echo "Checking if python is available"
+verify_python_version
+
+python3 --version

